### DISCUSSION TITLE
Distinguish Uno4/Uno5 at MSBuild level

### DISCUSSION THREAD

**Discussion by HavenDV** [unoplatform/uno#14332] [Q&A] [answered]
Hello

Do you have any properties that allow you to differentiate between Uno4 and Uno5 at the MSBuild level? This can be any constants or MSBuild property
This is necessary due to API changes to support Uno4/Uno5 at the same time in https://github.com/HavenDV/DependencyPropertyGenerator

**Comment by jeromelaban** [comment]
There isn't any, but you should be able to determine the version through the assembly informational version that contains `FrameworkElement`, for instance. That being said, it's probably best to alter your code based on features that you're expecting to see.

What's the change in Uno 5 that makes your generator alter its behavior?

**Reply by HavenDV** [reply]
The generator now specifies explicit names for parameters in the generated code to avoid API changes like this, and RegisterAttached parameter names have changed.
```
error CS1739: The best overload for 'RegisterAttached' does not have a parameter named 'typeMetadata' 
```
But I'll check to see if it's just one change and I can just remove the parameter name at this point.

**Reply by jeromelaban** [reply, marked as answer]
I see. thanks. You can also check for the signature of that particular method. It's likely more stable than looking for the Uno version.
